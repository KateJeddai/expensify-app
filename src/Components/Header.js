import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../Actions/auth';

export const Header = ({ startLogout }) => (
    <header>
      <h1>Expensify</h1>
      <NavLink to="/dashboard" activeClassName="is-active">Dashboard</NavLink>
      <p></p>
      <NavLink to="/create" activeClassName="is-active">Create Page</NavLink>
      <p></p>
      <NavLink to="/edit" activeClassName="is-active">Edit Page</NavLink>
      <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);