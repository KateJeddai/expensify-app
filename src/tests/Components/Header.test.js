import React from 'react';
import { shallow } from 'enzyme';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import toJSON from 'enzyme-to-json';
import Header from '../../Components/Header';

test('should render header', () => {
   /* const renderer = new ReactShallowRenderer();
    renderer.render(<Header />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();*/

    const wrapper = shallow(<Header />);
    expect(toJSON(wrapper)).toMatchSnapshot();

    //expect(wrapper.find('h1').length).toBe(1);
    //expect(wrapper.find('h1').text()).toBe('Expensify');



});