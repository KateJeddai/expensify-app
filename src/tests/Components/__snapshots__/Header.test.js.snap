// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <p />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create Page
  </NavLink>
  <p />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/edit"
  >
    Edit Page
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
